### PR TITLE
feat: added initial noop plugin

### PR DESCRIPTION
**Pull request by crutchcorn**
Just as mentioned in the docs, this is the plugin to do the noop and remove the "decorator". Code is pretty simple and I've ensured that we included tests.

I also added an `index.js` file so that the `will-mutate` import will work properly

### CODE REVIEW

File: noop/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+const {looksLike} = require("../utils/looks-like");
+
+
+module.exports = () => {
+	return {
+		name: "willMutate",
+		visitor: {
+			/**
+             * We need to traverse the "Program" so we can know if we need to inject
+             * the Proxy handler or not.
```
Comment (evelynhathaway) [inline noop/index.js:11]: Nit: spaces

Comment (evelynhathaway) [inline noop/index.js:11]: Is this comment accurately describing this program?

File: noop/__tests__/__fixtures__/basic/code.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+};
+
+$shouldNotMutate(["foo"]);
+function bar (foo, other) {
+	foo.prop = "Test";
+	other.prop = "Don't change me";
+}
+
+/**
+ * This does not currently work
```
Comment (evelynhathaway) [inline noop/__tests__/__fixtures__/basic/code.js:16]: Let's document this in a caveats section

File: index.js [hunk 3]
```diff
@@ -0,0 +1 @@
+module.exports = require("./plugin");
```
Comment (evelynhathaway) [inline index.js:1]: This is redundant with our `main` in our package.json. Either update the package.json if this is needed or remove this.

**Comment by evelynhathaway** [comment]
Question: Does this PR match [README - Babel Config](https://github.com/ceoss/will-mutate#babel-config)?

File: noop/index.js (hunk 1 above)
Comment (crutchcorn) [reply noop/index.js:11]: Legit no clue how this happened

Comment (crutchcorn) [reply noop/index.js:11]: Nooooope lol. Copy + paste hits again

File: index.js (hunk 3 above)
Comment (crutchcorn) [reply index.js:1]: Is it? I never understand the `main` in `package.json`

**Comment by crutchcorn** [comment]
@evelynhathaway yup! Should work the same way!

File: noop/__tests__/__fixtures__/basic/code.js (hunk 2 above)
Comment (crutchcorn) [reply noop/__tests__/__fixtures__/basic/code.js:16]: Let's, but not in this PR. This has existed in the codebase for some time and is unrelated to this change. FWIW,  this comment isn't accurate for the `noop` plugin, just the `plugin` one

File: noop/index.js (hunk 1 above)
Comment (evelynhathaway) [reply noop/index.js:11]: ESLint doesn't track `indent` on jsdoc-styled block comments IIRC.

File: noop/__tests__/__fixtures__/basic/code.js (hunk 2 above)
Comment (evelynhathaway) [reply noop/__tests__/__fixtures__/basic/code.js:16]: Ahh! Then I will break this out into another issue.

**Comment by crutchcorn** [comment]
Changes made, please re-review :)

**Pull request merged**